subprojects/core/src/main/java/org/gradle/api/internal/tasks/CachingTaskDependencyResolveContext.java

```diff
@@ -72,8 +66,13 @@ public Task getTask() {
         }
     }
 
-    private Set<Task> doResolve() {
-        walker.add(queue);
+    @Nullable
+    public Task getTask() {
+        return task;
+    }
+
+    private Set<Task> doGetDependencies(TaskDependency container) {
+        walker.add(container);
         return walker.findValues();
```

Comment:
It is a field since this is the `Caching` part of `CachingTaskDependencyResolveContext`.
The `walker` actually stores the tasks which have been examined as part of the task graph. That is why we keep adding things to it.
After every round of adding, we run `findValues()`, so the transitive hull of the added start nodes is discovered and the start nodes are cleared.